### PR TITLE
Discourage using dedicated staging environment

### LINKED ISSUE #240: Avoid creating additional environments?
> Create an additional `staging` environment that closely resembles the `production` one.

It sounds like this is advocating for adding a new config with `config/environments`. This is counter to the advice in https://12factor.net/config

> In a twelve-factor app, env vars are granular controls, each fully orthogonal to other env vars. They are never grouped together as “environments”, but instead are independently managed for each deploy. This is a model that scales up smoothly as the app naturally expands into more deploys over its lifetime.

From experience, when there are multiple environments (beyond the base 3), they often diverge, which lead to problems.

### PR DESCRIPTION
**Pull request by pirj**
Fixes #240

See:
- https://12factor.net/dev-prod-parity
- https://12factor.net/config

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -111,7 +111,8 @@ config.load_defaults 6.1
 
 === Staging Like Prod [[staging-like-prod]]
 
-Create an additional `staging` environment that closely resembles the `production` one.
+Avoid creating an additional `staging` environment, use `production`.
```
Comment (andyw8) [inline README.adoc:114]:
```suggestion
Avoid creating additional environment configurations than the defaults of `development`, `test` and `production`.
```

Comment (pirj) [reply README.adoc]: Thanks!

File: README.adoc [hunk 2]
```diff
@@ -111,7 +111,8 @@ config.load_defaults 6.1
...
+If certain configuration options for your staging server are different, configure them using environment variables.
```
Comment (andyw8) [inline README.adoc:115]:
```suggestion
If you need a production-like environment such as staging, use environment variables for configuration options.
```

File: README.adoc (hunk 1 above)
Comment (andyw8) [inline README.adoc]:
```suggestion
Avoid creating additional environment configurations other than the defaults of `development`, `test` and `production`.
```

**Comment by andyw8** [comment]
@pirj sorry, I meant to add my text as a suggestion rather than an edit.

**Comment by andyw8** [comment]
@pirj Oh, you merged so quickly, I didn't realise!

**Review by andyw8 — APPROVED** [review]

**Comment by pirj** [comment]
It doesn't take long to click a button 😄 
Thanks for the editing!

**Pull request merged**